src/lsm/grid.zig

```diff
@@ -207,7 +207,8 @@ pub fn GridType(comptime Storage: type) type {
             {
                 var it = grid.read_iops.iterate();
                 while (it.next()) |iop| {
-                    assert(address != iop.reads.peek().?.address);
+                    const iop_read = iop.reads.peek() orelse continue;
```

Comment:
Why did this change? I thought there was an invariant that a `read_iop` always has at least one `read` in its queue (the read that triggered the iop).

Reply:
I put the reasoning in the commits without as well being in the source (my mistake). This invariant is true up until [the last loop where the IOP's `read` FIFO is being dequeued](https://github.com/tigerbeetledb/tigerbeetle/blob/main/src/lsm/grid.zig#L417-L421). This last dequeue calls the read's callback which could enter one of the grid functions and iterate the IOPs again (expecting non-empty `read` FIFOs). I decided to forgo that invariant as the imagined control flow for handling the `release()` of the empty IOP before that last callback would've been awkward.